Source/x11/XGServerEvent.m

```diff
@@ -1912,27 +1912,40 @@ - (NSEvent *)_handleTakeFocusAtom: (XEvent)xEvent
                        forContext: (NSGraphicsContext *)gcontext
 {
   int key_num;
-  NSWindow *key_win;
+  NSWindow *keyWindow;
   NSEvent *e = nil;
-  key_win = [NSApp keyWindow];
-  key_num = [key_win windowNumber];
-  NSDebugLLog(@"Focus", @"take focus:%lu (current=%lu key=%d)",
-              cWin->number, generic.currentFocusWindow, key_num);
-
-  /* Sometimes window managers lose the setinputfocus on the key window
-   * e.g. when ordering out a window with focus then ordering in the key window.   
-   * it might search for a window until one accepts its take focus request.
-   */
-  if (key_num == cWin->number)
-    cWin->ignore_take_focus = NO;
+  gswindow_device_t *key_window;
+  
+  keyWindow = [NSApp keyWindow];
+  key_num = [keyWindow windowNumber];
+  key_window = [XGServer _windowWithTag:key_num];
```

Comment:
Could we move this line to 1974 as that structure only gets used there?

Reply:
No problem. Done.